src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-list/germplasm-list-creation.component.ts

```diff
@@ -81,6 +90,41 @@ export class GermplasmListCreationComponent implements OnInit {
         return this.nodes.push(this.toPrimeNgNode(node));
     }
 
+    onDrop(event, source: PrimeNgTreeNode, target: PrimeNgTreeNode) {
+        // Prevent to move source on same parent folder
+        if (source.parent.data.id === target.data.id) {
+            return;
+        }
+
+        // Prevent to move source if parent has a child with same name as the source
+        if (event.dropNode.children.find((node) => node.data.name === source.data.name)) {
+            this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.folder.move.parent.duplicated.name');
+            return;
+        }
+
+        if (source.children && source.children.length !== 0) {
+            this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.folder.cannot.move.has.children',
+                { folder: source.data.name });
+            return;
+        }
+
+        if (target.data.id === 'CROPLISTS' && !source.leaf) {
+            this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.folder.move.program.to.crop.list.not.allowed');
+            return;
+        }
+
+        if (target.leaf) {
+            this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.folder.move.not.allowed');
+            return;
+        }
+
+        event.accept();
+
+        this.treeService.move(source.data.id, target.data.id).subscribe((res) => {},
+            (res: HttpErrorResponse) =>
+                this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.error', { param: res.error.errors[0].message }));
+    }
```

Comment:
Can we can move the accept so that it works for server errors?
```typescript
this.treeService.move(source.data.id, target.data.id).subscribe(
     (res) => event.accept(),
     (res: HttpErrorResponse) => this.jhiAlertService.error('bmsjHipsterApp.tree-table.messages.error', { param: res.error.errors[0].message }
));
```